File: 01_Language/01_Functions/js/array_merge.js
function array_merge() {
    let res = []
    if (arguments.length === 0) {
        return []
    }

    for (let i in arguments) {
        if (!Array.isArray(arguments[i])) {
            res = {}
            break
        }
    }

    if (Array.isArray(res)) {
        res = res.concat(...arguments)
    } else {
        let index = 0
        for (let i in arguments) {
            for (let j in arguments[i]) {
                // typeof j === 'string'
                if (Number.isInteger(j * 1)) {
                    res[index] = arguments[i][j]
                    index++
                } else {
                    res[j] = arguments[i][j]
                }
            }
        }
    }
    return res
}

let array1 = {"color": "red", 0: 2, 1: 4}
let array2 = {0: "a", 1: "b", "color": "green", "shape": "trapezoid", 2: 4}
console.log(array_merge(array1, array2));

File: 01_Language/01_Functions/python/array_merge.py
# coding: utf-8

def array_merge(*args):
    res = []
    for i in args:
        if not isinstance(i, (list, tuple)):
            res = {}
            break
    if isinstance(res, list):
        for i in args:
            res += i
    else:
        index = 0
        for i in args:
            for k, v in i.items():
                if isinstance(k, int):
                    res[index] = v
                    index += 1
                else:
                    res[k] = v
    return res


if __name__ == '__main__':
    array1 = {"color": "red", 0: 2, 1: 4}
    array2 = {0: "a", 1: "b", "color": "green", "shape": "trapezoid", 2: 4}
    print(array_merge(array1, array2))
    print(array_merge([1, 2, 3], [4, 5, 6], [7, 8, 9]))
